remixer_core/src/main/java/com/google/android/libraries/remixer/Remixer.java

```diff
@@ -62,40 +62,39 @@ public static Remixer getInstance() {
    * This adds a remixer item ({@link Variable} or {@link Trigger}) to be tracked and displayed.
    * Checks that the remixer item is compatible with the existing remixer items with the same key.
    *
-   * <p>This method also removes old remixer items whose parent objects have been reclaimed by the
-   * Garbage collector which are being replaced by items from the same class of parent objects.
-   * No items are removed until equivalent ones from the same parent object class are added to
+   * <p>This method also removes old remixer items whose contexts have been reclaimed by the
+   * Garbage collector which are being replaced by items from the same class of context.
+   * No items are removed until equivalent ones from the same context class are added to
    * replace them. This guarantees that no incompatible items for the same key are ever accepted.
    *
    * @param remixerItem The remixer item to be added.
    * @throws IncompatibleRemixerItemsWithSameKeyException Other items with the same key have been
-   *     added by other parent objects with incompatible types.
-   * @throws DuplicateKeyException Another item with the same key was added by the same parent
-   *     object.
+   *     added other contexts with incompatible types.
+   * @throws DuplicateKeyException Another item with the same key was added for the same context.
    */
   @SuppressWarnings("unchecked")
   public void addItem(RemixerItem remixerItem) {
     List<RemixerItem> listForKey = getItemsWithKey(remixerItem.getKey());
     List<RemixerItem> itemsToRemove = new ArrayList<>();
     for (RemixerItem existingItem : listForKey) {
       existingItem.assertIsCompatibleWith(remixerItem);
-      if (!existingItem.hasParentObject()) {
-        // The parent activity has been reclaimed by the OS already. It has no callback and it's
+      if (!existingItem.hasContext()) {
+        // The context activity has been reclaimed by the OS already. It has no callback and it's
         // still around for keeping the value in sync, saving and checking consistency across types.
         // Since we're adding a new item that has been asserted to be compatible, it is not
         // necessary to keep this instance around.
         itemsToRemove.add(existingItem);
       } else {
-        // The parent activity is still alive and kicking.
-        if (existingItem.isParentObject(remixerItem.getParentObject())) {
-          // An object with the same key for the same parent object, this shouldn't happen so throw
+        // The context activity is still alive and kicking.
+        if (existingItem.isContext(remixerItem.getContext())) {
```

Comment:
is `isContext(Object)` still necessary? Can you just check `existingItem.getContext() == remixerItem.getContext()`?

If you need to keep it (because it does null checking), maybe rename it to `matchesContext()` or something.